File: src/prism.app/Models/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Prism.App.Models
{
    public class SocialPlayer
    {
        public int ID { get; set; }

        public int Level { get; set; }

        /// <summary>
        /// Experience is a summarization of all your Social Networks activity. 
        /// For example, each checkin at Foursquare gives you a experience points. Every single photo on a instagram, of course with likes and comments.
        /// everty tweet and so on.
        /// </summary>
        public long Exp { get; set; }

        /// <summary>
        /// This skill is only about likes and friends you have and related activity.
        /// </summary>
        public float SocialitySkill { get; set; }
        
        
        
        /// <summary>
        /// This skill is about your curiosity.
        /// Total kilometers and total diffirent places and average kilometers per one place.
        /// </summary>
        public float ExplorerSkill { get; set; }
    }
}

File: app/server/models/player.js
/* @flow */
var ExperienceConstants = require('./constants');

class SocialPlayer
{
  id: number;
  /// <summary>
  /// Experience is a summarization of all your Social Networks activity.
  /// For example, each checkin at Foursquare gives you a experience points.
  /// Every single photo on the instagram, of course with likes and comments.
  /// Every tweet and so on.
  /// </summary>
  exp: number;
  skills: any;
  achievements: any;
  userinfo: any;
  diagnose: any;
  level: number;

  constructor() {
    this.skills = [];
    this.exp = 0;
    this.level = 0;
  }

  addSkill(skill: any, points: number) {
    this.skills[skill] = points;
  }

  apply(skill: any, points: number, conditions?: bool) {

    if (points > 0 && conditions) {
      this.exp += points;
      this.level = Math.floor(Math.log(this.exp + 1, ExperienceConstants.LEVELFX_LOGARITHM_BASE) - 1);
      this.addSkill(skill, points);
    }
  }

}

//module.exports.PlayerSkill : PlayerSkill;
module.exports.Player = SocialPlayer;
